### ISSUE TITLE
No connection to port 1883

### ISSUE THREAD

**Issue by nicedevil007** [eclipse/mosquitto#2079]
heya, I run into an issue with my latest update on all my docker containers.
Everything is working except my eclipse-mosquitto one :(

I hope someone can help me out:

My docker stack yml looks like this:

```
version: "3.8"

volumes:
  mqtt_data:
  mqtt_config:
  mqtt_log:
  
networks:
 backend:

services:
  mqtt:
    image: eclipse-mosquitto
    hostname: mosquitto
    networks:
      backend:    
    ports:
      - 1883:1883/tcp
      - 9001:9001/tcp
    volumes:
      - mqtt_data:/mosquitto/data
      - mqtt_config:/mosquitto/config
      - mqtt_log:/mosquitto/log
    environment:
      TZ: "Europe/Vienna"
```

After the update to the lastest version I can't telnet to my host with port 1883 anymore.
What I found out so far is that any sent package to this container is going into the container (checked with `tcpdump -i any port 1883 -nn` inside the container) but nothing is coming back to the sending client of those packages.

This all was tested with freshly created volumes so the mosquitto.conf is the default one without any changes.
So connection via anonymous should be possilbe right?

I used mosquittoFX 1.7.1 to check if I even can connect to the broker without any luck :(

Hope someone can help me out?

**Comment by aureliosaraiva** [comment]
Same thing with me, I had to roll back to the previous version

**Comment by nicedevil007** [comment]
> Same thing with me, I had to roll back to the previous version

oh, I forgot to tell you that I tested a rollback to 1.16.2 and that didn't work as well now :(

Can you tell me how you fixed it with old version?

**Comment by missedtheapex** [comment]
I had the same issue when upgrading to 2.0.7 today (all default config). I tried rolling back and that didn't work.

I ran across this page: [http://www.steves-internet-guide.com/install-mosquitto-linux/](http://www.steves-internet-guide.com/install-mosquitto-linux/) , and it contained the following section:

> **Mosquitto 2..0.2 and above**
> Mosquitto v2 introduced some important changes that affect first time users in particular.
> 
> By default it requires authentication and doesn’t listen on a network address.
> 
> The following simple configuration file will make mosquitto start like previous versions:

```
listener 1883
allow_anonymous true
```
I uncommented the above lines in .../config/mosquitto.conf, except instead of "listener 1883" the uncommented line reads `port 1883`. And then my installation worked after restarting the container.

Seems like the documentation inside the conf file haven't been updated to explain how the defaults behave.

**Comment by nicedevil007** [comment]
> I had the same issue when upgrading to 2.0.7 today (all default config). I tried rolling back and that didn't work.
> 
> I ran across this page: http://www.steves-internet-guide.com/install-mosquitto-linux/ , and it contained the following section:
> 
> > **Mosquitto 2..0.2 and above**
> > Mosquitto v2 introduced some important changes that affect first time users in particular.
> > By default it requires authentication and doesn’t listen on a network address.
> > The following simple configuration file will make mosquitto start like previous versions:
> 
> ```
> listener 1883
> allow_anonymous true
> ```
> 
> I uncommented the above lines in .../config/mosquitto.conf, except instead of "listener 1883" the uncommented line reads `port 1883`. And then my installation worked after restarting the container.
> 
> Seems like the documentation inside the conf file haven't been updated to explain how the defaults behave.

ty for sharing this experience. my container keeps rebooting after setting `listener 1883` I have set a password with passwordfile so I use `allow_anonymous false`

**Comment by nicedevil007** [comment]
Ok fixed it as well now, what I did was the following (nearly the same as @missedtheapex did)

Removed the old container and all it's used volumes.

Then I created a new one with this docker stack... normal one from above, see my opening post.
Then I edited the `mosquitto.conf` as follows:

```
persistence true
persistence_location /mosquitto/data/

allow_anonymous false
password_file /mosquitto/config/credentials

listener 1883
```

Then I run into the shell of the container to create a new passwd file:

`mosquitto_passwd -c /mosquitto/config/credentials mqttuser`

And rebooted the container. Everything working again.
Thank you all for the tips to get it working :)
